inginious/frontend/pages/course_admin/submission_viewer.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+import pymongo
+import web
+import re
+import itertools
+import time
+import gettext
+from datetime import datetime
+from bson.objectid import ObjectId
+from collections import OrderedDict
+
+from inginious.frontend.pages.course_admin.utils import make_csv, INGIniousAdminPage
+from inginious.frontend.pages.course_admin.statistics import compute_statistics
+from inginious.common.base import id_checker
+
+class CourseSubmissionViewerTaskPage(INGIniousAdminPage):
+    """ List information about a task done by a student """
+    
+    def init(self, courseid):
+        course, __ = self.get_course_and_check_rights(courseid)
+        if course.is_lti():
+            raise web.notfound()
+            
+        self._allowed_sort = ["submitted_on", "username", "grade", "taskid"]
+        self._allowed_sort_name = [_("Submitted on"), _("User"), _("Grade"), _("Task id")]
+        self._valid_formats = ["taskid/username", "taskid/aggregation", "username/taskid", "aggregation/taskid"]
+        self._valid_formats_name = [_("taskid/username"), _("taskid/aggregation"), _("username/taskid"), _("aggregation/taskid")]
+        self._msg = [] # Put warning message here
+        self._trunc_limit = 1000 # To trunc submissions if there are too many submissions
+        self._time_alert_computation_stats = 2
+        
+        return course
+        
+    def POST(self, courseid):  # pylint: disable=arguments-differ
+        """ POST request """
+        course = self.init(courseid)
+
+        input = self.get_input()
+        tasks = course.get_tasks()
+        data, __ = self.get_submissions(course, input)
+
+        if "replay" in web.input():
+            if self.user_manager.has_admin_rights_on_course(course) == False:
+                raise web.notfound()
+            for submission in data:
+                self.submission_manager.replay_job(tasks[submission["taskid"]], submission)
+            self._msg.append(_("{0} selected submissions were set for replay.").format(str(len(data)))) 
+            
+        return self.page(course)
+
+    def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
+        """ GET request """
+        return self.page(self.init(courseid))
+
+    def submission_url_generator(self, submissionid):
+        """ Generates a submission url """
+        return "?submission=" + submissionid
+
+    def page(self, course):
+        """ Get all data and display the page """
+        
+        input = self.get_input()
```

Comment:
``input`` is the built-in user input method in Python, the variable should be renamed.